### ISSUE TITLE
Utilities to convert from option to SqlValue

### ISSUE THREAD

**Issue by njlr** [Zaid-Ajaj/Npgsql.FSharp#39]
Currently I am matching on my option types like this: 

```fsharp
type User = 
  {
    Email : string
    Name : string option
  }

let saveUser db user
  db
  |> Sql.query
    """
      INSERT INTO users ( email, name )
      VALUES ( @email, @name )
      RETURNING id
    """
  |> Sql.parameters
    [
      "email", Sql.Value user.Email
      "name", (
        match user.Name with
        | Some x -> SqlValue.String x
        | None -> SqlValue.Null
      )
    ]
  |> Sql.executeNonQueryAsync
```

Perhaps the library should provide something more concise like this? 

```fsharp
     Sql.MaybeValue user.Name
```

**Comment by Zaid-Ajaj** [comment]
Hello @njlr, 

That is not a bad idea, though instead of introducing a new function, we can have more overloads of the `Sql.Value` function that accept `Option<string>`, `Option<int>` etc and map those the either the value if it exists or `SqlValue.Null` when it doesn't. Would you be able to help out and send a PR? 

BTW, In your example above, you might want to replace `Sql.executeNonQueryAsync` with `Sql.executeScalar` because otherwise `executeNonQuery` will only give you the number of rows affected which will always be 1 if you insert a single user row

**Comment by njlr** [comment]
> Hello @njlr,
> 
> That is not a bad idea, though instead of introducing a new function, we can have more overloads of the `Sql.Value` function that accept `Option<string>`, `Option<int>` etc and map those the either the value if it exists or `SqlValue.Null` when it doesn't. Would you be able to help out and send a PR?

Yes happy to send a PR, although I cannot give you a timeline yet :)

> BTW, In your example above, you might want to replace `Sql.executeNonQueryAsync` with `Sql.executeScalar` because otherwise `executeNonQuery` will only give you the number of rows affected which will always be 1 if you insert a single user row

Yes, I copied and simplified some production code so it looks a little odd!

**Comment by ericsampson** [comment]
That would be handy :)

**Comment by njlr** [comment]
https://github.com/Zaid-Ajaj/Npgsql.FSharp/pull/40

**Comment by Zaid-Ajaj** [comment]
Fixed by #40
